online-shop/src/app/product-detail/product-detail.component.html

```diff
@@ -0,0 +1,33 @@
+<div *ngIf="product">
+  <h2>{{product.name | uppercase}} Details</h2>
+  <div><span>id: </span>{{product.id}} <br>
+       <span>Name: </span>{{product.name}} <br>
+       <span>Category: </span>{{product.category}} <br>
+       <span>Price: </span>{{product.price}} <br>
+       <span>Image: </span>{{product.image}} <br>
+       <span>Description: </span>{{product.description}} <br>
+
+  <br>
+  <button (click)="addToCart()">Add to cart</button><br><br>
+  <button> <a routerLink="/product-editor">Edit product</a></button><br><br>
+  <button (click)="deleteProduct()">Delete product</button><br><br><br><br><br><br><br><br><br><br><br><br><br>
+</div>
+  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
```

Comment:
you should handle the spaces with CSS, you can add a css class on the <div *ngIf="product"> 
and add a css rule like : "margin-bottom:40px;" or whatever space you need. same for all places with &nbsp